### PR TITLE
Republic Campaign Project

### PR DESCRIPTION
**Pull request by MasterOfGrey**
## **Republic Campaign**

This is the PR and collaboration thread for the Republic Campaign development. A separate project is in the works attempting to make a compelling campaign where you enlist in the Navy. However, it was felt by several people that this path was unviable, and also that the Republic deserved a campaign that wasn't intrinsically tied to military service. 

To that end, I worked with Warlord Mike who drafted the initial introductory content for the other campaign as its nominated writer before departing that project. Using the body of ground work created by Mike, and through many discussions with him, I have developed the following outline for a Republic story that we believe has great potential. (The underlying approach has also been endorsed by Amazinite and positive feedback received on the discord for the overarching themes.)

The outline is attached:
[republic storyboard6.docx](https://github.com/endless-sky/endless-sky/files/6875654/republic.storyboard6.docx)
Note: Alternative names for branches include "Investigator" and "War Hawk".

At present, the outline goes as far as the Pug Invasion.

Note: There are two "dump" files included in the files list for this PR. These files contain content referenced in the outline which is in need of being reworked to fit the new story path. They will be deleted once the relevant content is drawn across and reworked.

## How Will This Work?

@MasterOfGrey (That's Me): I will be taking care of the technical efforts of managing the branch, resolving conflicts, tracking plot areas where new features might be desired, keeping an eye out for plot contradictions or inconsistencies, and directing/helping people who want to contribute.

@WarlordMike : Mike is the lead writer, he played the greatest role in laying the foundations for this, and he intends to work on writing the main plot. He does not intend to work on scripting the missions around it though.

The following is an example of the format Mike is using for writing:
[Republic_Campaign_Script.pdf](https://github.com/endless-sky/endless-sky/files/6875787/Republic_Campaign_Script.pdf)

- If you are a scripter and want to help Mike or other writers turn it into game-friendly content, please speak up. It will be greatly appreciated. 
- If you are a writer and want to contribute, either by trying your hand at a section in the outline, developing an outline for the Pug invasion, or writing a side-plot. You are _**encouraged**_ to do so. Please post your intentions in a comment.
- If you want to do both, awesome! Welcome.

If you only do one and want to seek out someone to help you with the other feel free to consult the list below and ask them:

Scripters:
- @Terin 

Writers:
- @WarlordMike

I will script mission structures if necessary, but I am an unreliable person to ask for that at this point in time.

## What's to be done?

The following elements are required to bring this to completion:

1. [ ] Implementation of the Outline, with revisions as necessary.
2. [ ] Dump file content addressed in main and removed.
3. [ ] Plotting of the Pug Invasion -> Completion section.
4. [ ] Implementation of the Pug Invasion -> Completion section.
5. [ ] Adaptation of an electron beams mission chain.

The following elements are optional to bring this to completion:

6. [ ] Ideation of >3 side plots inc. ebeams.
7. [ ] Implementation of all side plots.
8. [ ] Possible "failure" branch, allowing the story to continue down a different path if you fail out of the main plot.
9. [ ] Possible "defection" branch.

## Save File
Any new save can be used.
_**Currently contains content to be replaced/reworked. Testing not recommended.**_

## PR Checklist
 - [ ] I updated the copyright attributions, or decline to claim copyright of any assets produced or modified
 - [ ] I uploaded the necessary image, blend, and texture assets here: {{insert link to assets}}
 - [ ] I created a PR to the [endless-sky-high-dpi repo](https://github.com/endless-sky/endless-sky-high-dpi) with the `@2x` versions of these art assets: {{insert PR link}}

### CODE REVIEW

File: data/human/republic dump intro.txt [hunk 1]
```diff
@@ -0,0 +1,1034 @@
...
+			`Your ship's communicator lights with an official announcement from the Republic:`
+			``
+			`	"Hello, Captain <last>," the message begins, "as of now, the Republic Navy is seeking captains skilled in combat to serve in the Navy. However, to discourage people hoping for handouts, applicants need to prove that they want to serve in the Navy. Recently, we've been having issues with a pirate vessel called the <npc> - our intel reports that it was spotted skulking around the <system> system. Destroy the <npc>, and return to <destination> to be formally accepted into the Navy."`
+			choice
+				`	I'll take on that ship.`
+					goto jobAccept
+				`	Sorry, but I have no desire to join the Navy at this time.`
+					defer
+			label jobAccept
+			`	"We look forward to seeing you on <destination> once the mission is over," the message reads. "Good luck!"`
```
Comment (TheMarksman-ES) [inline data/human/republic dump intro.txt:28]: You may want to consider the usage of "Good Hunting." It has some significance in the US military and would likely fit this much better.

File: data/human/republic dump intro.txt [hunk 2]
```diff
@@ -0,0 +1,1034 @@
...
+#		event "naval employment earned"
+		event "Navy Painted"
+		log "Was able to destroy the pirate ship and prove my worth to the Navy. The Navy has inducted me as an Ensign and given me a small salary."
+		conversation
+			`As you cycle the hatch to exit the ship, you are greeted by two Navy personnel flanking the entryway, armed with rifles. Between them, a Navy officer stands with his arms folded in front of him.`
+			``
+			`	"I assume you are <first> <last>?" he asks. You nod, saluting with your right hand. "Yes sir. And you are...?"`
+			`	"Colonel Marcus Howell," he replies. "Well met, Captain <last> - or rather, Ensign <last>."`
+			`	You nod, still saluting. "Thank you, sir. I hope to be of great service to the Republic."`
+			`	"I have a feeling you'll be just the pilot we need, <last>," he states while making a slight bounce on the tips of his feet, smiling. "The speed and skill with which your ship, the <ship>, destroyed that Raven, in addition to your past endeavors, proves you are both serious about joining the Navy... and it's ready for battle."`
```
Comment (TheMarksman-ES) [inline data/human/republic dump intro.txt:55]: "proves you are both serious about joining the Navy... and it's ready for battle."
You're both serious and... what, exactly? You're talking about first last and suddenly, that derailed.

File: data/human/republic dump intro.txt [hunk 3]
```diff
@@ -0,0 +1,1034 @@
...
+			``
+			`Howell's expression hardens as his gaze turns left and skyward. "With rumblings of war starting in the South, and pirates massing in the North, the Navy is going to be stretched thin." His gaze returns to you. "As of today, you haved been approved for Navy Tier I Enlistment status. As part of being inducted as a Navy Ensign, you'll receive a daily stipend of 500 credits per day. At ease."`
```
Comment (TheMarksman-ES) [inline data/human/republic dump intro.txt:57]:
```suggestion
			"With the rumblings of war growing in the South, and the massing of pirates in the North,
```

File: data/human/republic dump intro.txt [hunk 4]
```diff
@@ -0,0 +1,1034 @@
...
+			``
+			`	You let your hand drop to your side as you nod firmly. "What are my orders, Colonel?"`
```
Comment (TheMarksman-ES) [inline data/human/republic dump intro.txt:59]: Perhaps "Sir" here?

Comment (TheMarksman-ES) [reply data/human/republic dump intro.txt:59]: Since naming someone's rank without their name seems to be more common when a superior is talking to a subordinate.

File: data/human/republic dump intro.txt [hunk 5]
```diff
@@ -0,0 +1,1034 @@
...
+			`	"Your first assignment," he begins, taking a uniform handed to him by the trooper to his left (your right), "will be to rendezvous with Admiral William Danforth - he's been tasked to deal with the pirates that are starting to gather up North."`
+			`He hands you the uniform - it's the familiar blue Navy jumpsuit, speckled with orange and gold highlights here and there. Your hands are trembling slightly as you take the uniform from his hands - *this is really happening,* you think to yourself.`
```
Comment (TheMarksman-ES) [inline data/human/republic dump intro.txt:61]: Maybe emphasise whatever the rank insignia looks like?

File: data/human/republic dump intro.txt [hunk 6]
```diff
@@ -0,0 +1,1034 @@
...
+			`	...and Hatysa."`
+			``
+			`Several of the enlisted present quietly chatter amongst themselves in various tones. "I cannot emphasize this enough, ladies and gentlemen," Danforth warns. "With all the ships we lost in the nuclear attack, we simply cannot afford to lose more ships. Each of you will report to the base on <destination> for your first assignment. Any questions?"`
+			``
+			`	A long, awkward silence falls over the room. Danforth gazes over the group a few times before a small smile is visible on his face.`
+			``
+			`"Good. You have your orders, men - dismissed!"`
+				accept
+	on accept
+		log "Have reported to Admiral Danforth for patrol duty against pirates."
```
Comment (TheMarksman-ES) [inline data/human/republic dump intro.txt:102]: Drop the Have.

**Review by TheMarksman-ES — CHANGES_REQUESTED** [review]
Leaving it as that for now.

**Comment by kestrel1110** [comment]
So is everyone gonna go with this version where you work for the Republic instead of enlisting in the Navy or can I still finish the enlistment storyline?

**Comment by BlazingDiesel** [comment]
I'm particularly interested in scripting for the suggested additional intro (intelligence) mission chain. Considering my somewhat limited experience, I feel that a task like this would suit what I feel I'm capable of.

As of right now I've read the other current into missions in their entirety, as well as about half of the intended storyboard, and seeing that there will indeed be connections from this chain later on is helpful. I'm a bit curious how this mission chain would play around with the others, with the idea out there that it may/may not require the scouting missions and whatever else. Would be helpful to know, but again I am a little inexperienced so if there's someone more capable at this kind of thing who can help I won't complain.

**Comment by MasterOfGrey** [comment]
> So is everyone gonna go with this version where you work for the Republic instead of enlisting in the Navy or can I still finish the enlistment storyline?

There is no restriction on submitting storylines, I did say there's a separate project in the works for an enlistment plot so you're welcome to keep going with that. We just didn't think that that plot was as workable as this one so we've elected to go this way.

> I'm particularly interested in scripting for the suggested additional intro (intelligence) mission chain. Considering my somewhat limited experience, I feel that a task like this would suit what I feel I'm capable of.
> 
> As of right now I've read the other current into missions in their entirety, as well as about half of the intended storyboard, and seeing that there will indeed be connections from this chain later on is helpful. I'm a bit curious how this mission chain would play around with the others, with the idea out there that it may/may not require the scouting missions and whatever else. Would be helpful to know, but again I am a little inexperienced so if there's someone more capable at this kind of thing who can help I won't complain.

Heya, there's a lot of scope for more intro missions. Especially given that the Republic is where everyone starts I personally think it would be best if the volume of intro missions you had to complete was higher, so that new players aren't pushed into the Republic plot too swiftly.

I think that requiring completion of at least some of the scouting missions is a pretty fair. You don't need to be overly concerned about having extra intro content interact with things specifically. Just having more intro content gives us more scope and more things to use. 

Basically, if you want to make more intro content, we'll make it work somewhere.

**Comment by samoja12** [comment]
> > So is everyone gonna go with this version where you work for the Republic instead of enlisting in the Navy or can I still finish the enlistment storyline?
> 
> There is no restriction on submitting storylines, I did say there's a separate project in the works for an enlistment plot so you're welcome to keep going with that. We just didn't think that that plot was as workable as this one so we've elected to go this way.
> You can still do the enlistment storyline, nobody will stop you. :)
> 
> > I'm particularly interested in scripting for the suggested additional intro (intelligence) mission chain. Considering my somewhat limited experience, I feel that a task like this would suit what I feel I'm capable of.
> > As of right now I've read the other current into missions in their entirety, as well as about half of the intended storyboard, and seeing that there will indeed be connections from this chain later on is helpful. I'm a bit curious how this mission chain would play around with the others, with the idea out there that it may/may not require the scouting missions and whatever else. Would be helpful to know, but again I am a little inexperienced so if there's someone more capable at this kind of thing who can help I won't complain.
> 
> Heya, there's a lot of scope for more intro missions. Especially given that the Republic is where everyone starts I personally think it would be best if the volume of intro missions you had to complete was higher, so that new players aren't pushed into the Republic plot too swiftly.
> 
> I think that requiring completion of at least some of the scouting missions is a pretty fair. You don't need to be overly concerned about having extra intro content interact with things specifically. Just having more intro content gives us more scope and more things to use.
> 
> Basically, if you want to make more intro content, we'll make it work somewhere.

Personally i would prefer a branching point where you choose to either enlist or remain a contractor, that way you could have one intro and the whole thing would feel less disjointed, but if you think that's unworkable then just do as you think is right.

**Comment by MasterOfGrey** [comment]
> Personally i would prefer a branching point where you choose to either enlist or remain a contractor, that way you could have one intro and the whole thing would feel less disjointed, but if you think that's unworkable then just do as you think is right.

_**If**_ a viable enlistment storyline appears I'm happy to coordinate a link between them. Bereskatuket/kestrel110 is certainly working on one and I wish them the best of luck.
I don't personally believe that such a storyline is viable though, so I won't be going out of my way to accommodate for it before it happens.

File: data/human/republic dump intro.txt (hunk 1 above)
Comment (roadrunner56) [reply data/human/republic dump intro.txt:28]: Hmm, but would a galactic republic remember or care about the military traditions of a 1000 year old country?

Comment (MasterOfGrey) [reply data/human/republic dump intro.txt:28]: This review comment is on the dump file, i.e. the file where everything in it is being reworked and replaced...

File: data/human/republic start.txt [hunk 7]
```diff
@@ -0,0 +1,251 @@
...
+			choice
+				`	"I'll join the Republic. What's our next step?"`
+					goto join
+				`	"Sorry, but I'm partial to the Free World's cause."`
+				`	"Sorry, but I'd prefer not to pick a side in this."`
+			`	Colonel Howell sighs. "I understand you've made your decision. It's not an easy thing to commit to either side, especially in a conflict like this." You say goodbye to Colonel Howell and Commander Melgrew, and then leave the office.`
+				decline
+			label join
+			`	"Good," Howell says. "We'll get started, then. For the time being, you'll report to Jane. She needs some reconaissance done in systems with substantial pirate activity. You two are dismissed." You leave his office with Melgrew and make your way to the spaceport.`
+			`	"First on out list, like Marcus said, is scouting. I need you pulling flybys on three systems: <waypoints>. Report back to me with your sensor logs when you're done. Oh, and your payment is <payment>, which should only go up from here. Got it?"`
```
Comment (warp-core) [inline data/human/republic start.txt:139]:
```suggestion
			`	"First on our list, like Marcus said, is scouting. I need you pulling flybys on three systems: <waypoints>. Report back to me with your sensor logs when you're done. Oh, and your payment is <payment>, which should only go up from here. Got it?"`
```

File: data/human/republic start.txt [hunk 8]
```diff
@@ -0,0 +1,251 @@
...
+	name "Pirate Patrol"
+	description "Fly through <waypoints> before returning to <origin> to deliver your sensor logs."
+	waypoint "Arneb"
+	waypoint "Alnilam"
+	waypoint "Hatysa"
+	source "Farpoint"
+	on fail
+		dialog `You have failed an essential Republic mission. If you want to complete the story line, revert to the autosave or another earlier snapshot of the game.`
+	to offer
+		has "Pirate Patrol 1: done"
```
Comment (warp-core) [inline data/human/republic start.txt:186]:
```suggestion
		has "Republic Pirate Patrol 1: done"
```

**Review by warp-core — COMMENTED** [review]
I think, for future reference, all Republic missions should begin with 'Republic', not sure about if there should be a colon or not after it.
